tfx/components/example_gen/write_split.py

```diff
@@ -54,14 +58,25 @@ def WriteSplit(
     example_split: beam.pvalue.PCollection,
     output_split_path: str,
     output_format: str,
+    output_payload_format: int,
 ) -> beam.pvalue.PDone:
   """Shuffles and writes output split as serialized records in TFRecord."""
   del output_format
 
-  return (example_split
+  shuffled = (example_split
           | 'MaybeSerialize' >> beam.ParDo(MaybeSerialize())
           # TODO(jyzhao): make shuffle optional.
-          | 'Shuffle' >> beam.transforms.Reshuffle()
+          | 'Shuffle' >> beam.transforms.Reshuffle())
+
+  if output_payload_format == example_gen_pb2.PayloadFormat.FORMAT_PARQUET:
+    # TODO: propagate schema.
+    schema = pyarrow.schema(list())
```

Comment:
write parquet needs a schema, which we don't have access to, unless we propagate the schema when we read the parquet files.

Reply:
can we put the schema in exec_properties when reading and use it here?